### PR TITLE
Fix dependency of geant4 (amends #16497)

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
Let me actually wait for maintainer approval this time instead of merging things quickly

**Review by drbenmorgan — CHANGES_REQUESTED** [review]
The 10.6.1 patch also requires g4emlow 7.9.1 patch, so that version needs adding to the g4emlow `package.py`, and the `depends_on` in `geant4-data` updating accordingly.

File: var/spack/repos/builtin/packages/geant4-data/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4-data/package.py]:
```suggestion
    depends_on("g4emlow@7.9", when='@10.6.0')
```

**Review by drbenmorgan — APPROVED** [review]

**Pull request merged**
